client/engine/chainservice/mockchain.go

```diff
@@ -0,0 +1,71 @@
+package chainservice
+
+import (
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// MockChain provides an interface which simulates a blockchain network. It is designed for use as a central service which multiple
+// ChainServices connect to with Go chans.
+// It keeps a record of of holdings and adjudication status for each channel, accepts transactions and emits events.
+type MockChain struct {
+	out map[types.Address]chan Event // out is a mapping with a chan for each connected ChainService, used to send Events to that service
+	in  chan protocols.Transaction   // in is the chan used to recieve Transactions from multiple ChainServices
+
+	holdings map[types.Destination]types.Funds // holdings tracks funds for each channel
+}
+
+// Out() returns the out chan for a particular ChainService, and narrows the type so that external consumers mays only recieve on it.
+func (mc MockChain) Out(a types.Address) <-chan Event {
+	return mc.out[a]
+}
+
+// In returns the in chan but narrows the type so that external consumers may only send on it.
+func (mc MockChain) In() chan<- protocols.Transaction {
+	return mc.in
+}
+
+// NewMockChain returns a new MockChain with an out chan initialized for each of the addresses passed in.
+func NewMockChain(addresses []types.Address) MockChain {
+
+	mc := MockChain{}
+	mc.out = make(map[types.Address]chan Event)
+	mc.in = make(chan protocols.Transaction)
+	mc.holdings = make(map[types.Destination]types.Funds)
+
+	const BUFFER_SIZE = 10
```

Comment:
The test passes with no set buffer. As I read `handleTx`, a non-reading bob will only block further events on his own SimpleMessageService, but not for other engines.

Is this right? Is this buffer required?

Reply:
Yes you're right. I think this was a hangover from a debugging session where I hadn't yet made a goroutine for each send from the `MockChain`.